### ISSUE TITLE
Bar charts with multiple colors renders as all black in safari, works in chrome

### ISSUE THREAD

**Issue by LACoderDeBH** [recharts/recharts#1411]
## What is the current behavior?
recharts Composed charts is correctly rendering my different colors (different values) in Chrome. In safari it renders as all black bars. 

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via 
Chrome:
![screen shot 2018-07-31 at 4 02 59 pm](https://user-images.githubusercontent.com/28251592/43491936-6acf3fb8-94db-11e8-8776-058316517940.png)

Safari (Not working):
![screen shot 2018-07-31 at 4 02 40 pm](https://user-images.githubusercontent.com/28251592/43491945-6e2b8c84-94db-11e8-9cd4-0bb8a059605e.png)

## What is the expected behavior?
Have safari render bar graph in composed chart the same as chrome.

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
1.0.1

**Comment by LACoderDeBH** [comment]
Figured it out. Conditional rendering in the fill field using rgba values was causing an issue. I replaced the fill with just one option and it works.

**Comment by wrenzi** [comment]
Hey folks. I solved a similar Issue, and I'll share the problem/solution with you.

The problem, in our case, was not with the recharts themselves, but with our implementation.

We were adding two graphs to our page, with their respectives `defs`, and the `defs` were duplicating (two `defs` with the same id). The issue happened when we removed the last `linearGradient` def, and kept the first one. It made our `AreaChart` lose the fill def reference, even though the first one still exists.

The solution was to create `linearGradient` with different ids.

I hope it helps someone.
